sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client.c

```diff
@@ -12,9 +12,160 @@
+
+  ret = az_iot_provisioning_client_user_name_get(&client, username, &username);
+  assert_int_equal(AZ_OK, ret);
+  assert_memory_equal(
+      az_span_ptr(expected_username), az_span_ptr(username), (size_t)az_span_length(username));
+  assert_int_equal(0xCC, username_buffer[az_span_length(username)]);
+}
+
+static void test_az_iot_provisioning_client_get_subscribe_topic_filter_succeed()
+{
+  az_iot_provisioning_client client;
+  uint8_t topic_buffer[128];
+  az_span topic = AZ_SPAN_LITERAL_FROM_BUFFER(topic_buffer);
+  az_span_set(topic, 0xCC);
+
+  az_span expected_topic = AZ_SPAN_LITERAL_FROM_STR("$dps/registrations/res/#");
+
+  az_result ret
+      = az_iot_provisioning_client_register_subscribe_topic_filter_get(&client, topic, &topic);
+  assert_int_equal(AZ_OK, ret);
+  assert_memory_equal(
+      az_span_ptr(expected_topic), az_span_ptr(topic), (size_t)az_span_length(topic));
+
+  assert_int_equal(0xCC, topic_buffer[az_span_length(topic)]);
+}
+
+static void test_az_iot_provisioning_client_get_register_publish_topic_filter_succeed()
+{
+  az_iot_provisioning_client client;
+  uint8_t topic_buffer[128];
+  az_span topic = AZ_SPAN_LITERAL_FROM_BUFFER(topic_buffer);
+  az_span_set(topic, 0xCC);
+
+  az_span expected_topic
+      = AZ_SPAN_LITERAL_FROM_STR("$dps/registrations/PUT/iotdps-register/?$rid=1");
+
+  az_result ret = az_iot_provisioning_client_register_publish_topic_get(&client, topic, &topic);
+  assert_int_equal(AZ_OK, ret);
+  assert_memory_equal(
+      az_span_ptr(expected_topic), az_span_ptr(topic), (size_t)az_span_length(topic));
+  assert_int_equal(0xCC, topic_buffer[az_span_length(topic)]);
+}
+
+static void test_az_iot_provisioning_client_get_operation_status_publish_topic_filter_succeed()
+{
+  az_iot_provisioning_client client;
+  uint8_t topic_buffer[128];
+  az_span topic = AZ_SPAN_LITERAL_FROM_BUFFER(topic_buffer);
+  az_span_set(topic, 0xCC);
+
+  az_span expected_topic
+      = AZ_SPAN_LITERAL_FROM_STR("$dps/registrations/GET/iotdps-get-operationstatus/"
+                                 "?$rid=1&operationId=4.d0a671905ea5b2c8.42d78160-4c78-479e-8be7-"
+                                 "61d5e55dac0d" TEST_REQUEST_ID);
+
+  az_span operation_id
+      = AZ_SPAN_LITERAL_FROM_STR("4.d0a671905ea5b2c8.42d78160-4c78-479e-8be7-61d5e55dac0d");
+  az_iot_provisioning_client_register_response response = { 0 };
+  response.operation_id = operation_id;
+
+  az_result ret = az_iot_provisioning_client_get_operation_status_publish_topic_get(
+      &client, &response, topic, &topic);
+
+  assert_int_equal(AZ_OK, ret);
+  assert_memory_equal(
+      az_span_ptr(expected_topic), az_span_ptr(topic), (size_t)az_span_length(topic));
+  assert_int_equal(0xCC, topic_buffer[az_span_length(topic)]);
+}
+
+#ifdef _MSC_VER
+// warning C4113: 'void (__cdecl *)()' differs in parameter lists from 'CMUnitTestFunction'
+#pragma warning(disable : 4113)
```

Comment:
Do we want to re-enable the warning outside this file?

Reply:
I think we need to unify this into something that everybody can use: I don't like having to write and read useless `**state` parameters in all our tests. 

> outside this file

My understanding is that the preprocessor rules apply for this file only and that `#pragma warning` applies only from this line within the file (not above).